File: frontend/src/types/unidadMedida.ts
export interface UnidadMedida {
  id: number;
  nombre: string;
  simbolo: string;
  tipo: string;
  activo: boolean;
  created_at: string;
  updated_at: string;
  deleted_at?: string | null;
}

export interface UnidadMedidaPayload {
  nombre: string;
  simbolo: string;
  tipo: string;
  activo?: boolean;
}

File: backend/app/models/unidad_medida.py
from __future__ import annotations

from datetime import datetime, timezone
from typing import Optional

from sqlmodel import Field, SQLModel


class UnidadMedida(SQLModel, table=True):
    __tablename__ = "unidad_medida"

    id: Optional[int] = Field(default=None, primary_key=True)
    nombre: str = Field(index=True, unique=True, min_length=2, max_length=50)
    simbolo: str = Field(index=True, unique=True, min_length=1, max_length=10)
    tipo: str = Field(min_length=2, max_length=20)
    activo: bool = Field(default=True)
    created_at: datetime = Field(default_factory=lambda: datetime.now(timezone.utc))
    updated_at: datetime = Field(default_factory=lambda: datetime.now(timezone.utc))
    deleted_at: Optional[datetime] = Field(default=None)
